modules/core/src/main/java/org/apache/ignite/internal/processors/metastorage/persistence/DistributedMetaStorageImpl.java

```diff
@@ -1070,6 +1125,31 @@ else if (!isClient && ver.id() > 0) {
         return fut;
     }
 
+    /** */
+    private GridFutureAdapter<Boolean> validateBeforeWrite(String key) throws IgniteCheckedException {
+        boolean supported;
+
+        try {
+            supported = isSupported(ctx);
```

Comment:
Is it really necessary to validate all the nodes on every operation?
Do we allow old nodes to connect to new grid? If no, then may be we can skip this check if once found all the nodes support the feature?

Reply:
Every component does this, changing the approach is out of the scope. And yes, older node can enter the cluster sometimes.